emulator.c

```diff
@@ -785,6 +806,51 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
         num_cycles = MOV_FROM_MEM(cpu, &cpu->a);
         break;
       }
+    case 0x80: // NOLINT
+      {        // ADD B
+        num_cycles = add_reg_accum(cpu, cpu->b);
+        break;
+      }
+    case 0x81: // NOLINT
+      {        // ADD C
+        num_cycles = add_reg_accum(cpu, cpu->c);
+        break;
+      }
+    case 0x82: // NOLINT
+      {        // ADD D
+        num_cycles = add_reg_accum(cpu, cpu->d);
+        break;
+      }
+    case 0x83: // NOLINT
+      {        // ADD E
+        num_cycles = add_reg_accum(cpu, cpu->e);
+        break;
+      }
+    case 0x85: // NOLINT
+      {        // ADD L
+        num_cycles = add_reg_accum(cpu, cpu->l);
+        break;
+      }
+    case 0x86: // NOLINT
+      {        // ADD M
+        uint8_t value = cpu_read_mem(cpu, cpu->pc + 1);
```

Comment:
I don't believe this is correct. M is (HL) not an immediate. 

num_cycles = ADD(cpu, cpu_read_mem(cpu, readRegisterPair(cpu, HL))); 

Should work and allow you to use your helper function.